web/crux-ui/src/pages/projects.tsx

```diff
@@ -51,7 +51,18 @@ const ProjectsPage = (props: ProjectsPageProps) => {
   const [creating, setCreating] = useState(false)
   const submitRef = useRef<() => Promise<any>>()
 
-  const [viewMode, setViewMode] = useState<ViewMode>('tile')
+  const [viewMode, setViewMode] = useState<ViewMode>(() => {
```

Comment:
I think it'd be better to implement a custom hook which would manage the ViewMode as well as save it to local storage. It could be something like `usePersistedViewMode(name)`. And in this case the name would be 'projects' so later (if we need to) we could use it somewhere else too.

Reply:
We got a new usePersistedViewMode custom hook for this purpose, which is capable of handling view modes specific to each page. 
We now use this on the Projects, deployment state, and version state pages as well.